src/blindpsbt.cpp

```diff
@@ -400,83 +400,40 @@ BlindingStatus BlindPSBT(PartiallySignedTransaction& psbt, std::map<uint32_t, st
                     GenerateAssetEntropy(entropy, input.GetOutPoint(), input.m_issuance_asset_entropy);
                 }
 
-                // Asset isn't blinded yet. Add it to the list of input assets
-                CalculateAsset(issuance_asset, entropy);
-                fixed_input_tags.emplace_back();
-                memcpy(fixed_input_tags.back().data, issuance_asset.begin(), 32);
-                ephemeral_input_tags.emplace_back();
-                if (secp256k1_generator_generate(secp256k1_blind_context, &ephemeral_input_tags.back(), issuance_asset.begin()) != 1) {
-                    return BlindingStatus::INVALID_ASSET;
+                if (input.m_issuance_value != std::nullopt || input.m_issuance_value_commitment.IsCommitment()) {
+                    // Asset isn't blinded yet. Add it to the list of input assets
+                    CalculateAsset(issuance_asset, entropy);
+                    fixed_input_tags.emplace_back();
+                    memcpy(fixed_input_tags.back().data, issuance_asset.begin(), 32);
+                    ephemeral_input_tags.emplace_back();
+                    if (input.m_issuance_value_commitment.IsNull()) {
+                        if (secp256k1_generator_generate(secp256k1_blind_context, &ephemeral_input_tags.back(), issuance_asset.begin()) != 1) {
```

Comment:
I believe `&ephemeral_input_tags.back()` should be `ephemeral_input_tags.back().data`.

Reply:
Ah, no, `ephemeral_input_tags.back()` returns a `secp256k1_generator`. The real bug here is that we ever touch `.data` which is supposed to be a secp-zkp implementation detail, but that's a fight for another day.